### PR TITLE
draft PR-2

### PR DESCRIPTION
**Pull request by OkoliEvans**
- [ ] issue #
- [ ] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

### CODE REVIEW

File: onchain/src/tests/art_peace.cairo [hunk 1]
```diff
@@ -428,3 +428,75 @@ fn deposit_reward_test() {
...
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
+
+    start_prank(CheatTarget::One(art_peace_address), user3);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
+
+    start_prank(CheatTarget::One(art_peace_address), user4);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
```
Comment (ptisserand) [inline onchain/src/tests/art_peace.cairo:489]: All users are setting the same pixel and the color is the wrong one.
You should use these values `(pos, color)`:
```
(0, 1)
(1, 2)
(WIDTH, 3)
(WIDTH + 1, 4)
```

File: onchain/src/art_peace.cairo [hunk 2]
```diff
@@ -982,7 +1004,31 @@ pub mod ArtPeace {
...
+                            let user_reward = (reward_amount * user_total_pixels.into()) / total_pixels.into();
+
+                            if user == Zero::zero() {
+                                panic!("no user recorded"); // for testing purpose only
+                            } else {
+                                let success = IERC20Dispatcher { contract_address: reward_token }
+                                    .transfer(user, user_reward);
+                                assert(success, 'ERC20 transfer fail!');
+                            }
+                        }
```
Comment (ptisserand) [inline onchain/src/art_peace.cairo:1030]: Inifinite loop, you need to add:
```
i += 1;
```

File: onchain/src/art_peace.cairo [hunk 3]
```diff
@@ -982,7 +1004,31 @@ pub mod ArtPeace {
...
+                        .len() {
+                            let reward_token = template_metadata.reward_token;
+                            let reward_amount = template_metadata.reward;
+                            let total_pixels = self.total_pixels.read();
+
+                            let mut user = *pixel_contributors.at(i).into();
+                            let user_index = (i + 1);
+                            let user_total_pixels = total_pixels_by_user.get(user_index.into());
+                            
+                            let user_reward = (reward_amount * user_total_pixels.into()) / total_pixels.into();
```
Comment (ptisserand) [inline onchain/src/art_peace.cairo:1021]: `total_pixels` is the number of pixels in canvas.
You have to divide with number of pixels in template, i.e.: `template_metadata.width * template_metadata.height`

**Comment by OkoliEvans** [comment]
@ptisserand  pls help me check if there are still any errors, test still doesn't pass on my machine

**Pull request closed without merging**